### PR TITLE
fix: Set proper `notfound_urls_prefix` option types

### LINKED ISSUE #134: Disabling prefix leads to "WARNING: The config value `notfound_urls_prefix' has type `NoneType', defaults to `str'."
**Python version:** 3.9.0
**Sphinx version:** 1.8.5
**sphinx-notfound-page version:** 0.6

Here's the relevant part of my `conf.py`:

```python
# sphinx-notfound-page
# https://github.com/readthedocs/sphinx-notfound-page
notfound_context = {
    "title": "Page Not Found",
    "body": """
<h1>Page Not Found</h1>
<p>Sorry, we couldn"t find that page.</p>
<p>Try using the search box or go to the homepage.</p>
""",
}

# on_rtd is whether we are on readthedocs.org, this line of code grabbed from docs.readthedocs.org
on_rtd = os.environ.get("READTHEDOCS", None) == "True"

# Don't add `/en/latest` prefix during local development.
# This makes it easier to test the custom 404 page by loading `/404.html`
# on a local web server.
if not on_rtd:
    notfound_urls_prefix = None
```

I've been able to avoid this by setting the value to `''` (empty string) instead, but the documentation doesn't mention this. I guess we should just update the documentation to reflect this?

### PR DESCRIPTION
**Pull request by BoboTiG**
Fixes #134.

I added a regression test that fails before that patch, and passes  with the current patch applied.

### CODE REVIEW

File: tests/test_urls.py [hunk 1]
```diff
@@ -39,6 +39,13 @@ def test_parallel_build():
     # https://github.com/sphinx-doc/sphinx/pull/8257
     subprocess.check_call('sphinx-build -j 2 -W -b html tests/examples/parallel-build build', shell=True)
 
+
+@pytest.mark.sphinx(srcdir=srcdir)
+def test_notfound_urls_prefix_can_be_none():
+    """Note: Sphinx warnings being treated as errors, if the test contains any warning, it will fail then."""
+    subprocess.check_call('sphinx-build -W -b html tests/examples/conf-urls-prefix-none build', shell=True)
```
Comment (humitos) [inline tests/test_urls.py:46]: The usage of `subprocess` in the previous tests was kind of a hack for parallel building, but it shouldn't be required for this particular test.

You can follow as example the test `test_context_settings` that uses `confoverrides` to change a setting and define `notfound_urls_prefix=None` there.

File: tests/examples/conf-urls-prefix-none/conf.py
Comment (humitos) [inline tests/examples/conf-urls-prefix-none/conf.py:1]: We shouldn't require a new example test folder for this test. Take a look at how other tests use the `confoverride` parameter to override a configuration in the Sphinx test environment.

**Review by humitos — CHANGES_REQUESTED** [review]
Looks good to me! 💯 

Thanks for opening this PR. The only change it's required here is to follow the same pattern we are using via `confoverrides` and don't creating another folder for this test case. I'd appreciate if you can change that so we can move forward and merge this PR 🙏🏼

File: tests/examples/conf-urls-prefix-none/conf.py
Comment (BoboTiG) [reply tests/examples/conf-urls-prefix-none/conf.py:1]: Nice! The new version is way lighter!

**Review by humitos — APPROVED** [review]
Excellent!

**Pull request merged**